dcos/http.py

```diff
@@ -1,11 +1,19 @@
+import getpass
+import sys
+
 import requests
 from dcos import util
 from dcos.errors import DCOSException, DCOSHTTPException
+from requests.auth import HTTPBasicAuth
+
+from six.moves.urllib.parse import urlparse
 
 logger = util.get_logger(__name__)
 
 DEFAULT_TIMEOUT = 5
 
+AUTH_CREDS = {}  # (hostname, realm) -> AuthBase()
```

Comment:
changed a few things around so it should be safe according to this: http://effbot.org/pyfaq/what-kinds-of-global-value-mutation-are-thread-safe.htm